src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/RetryApiService.kt

```diff
@@ -29,12 +30,13 @@ class RetryApiService(
       .filter { isTimeoutException(it) }
       .doBeforeRetry { logRetrySignal(it) }
 
-  private fun isTimeoutException(it: Throwable): Boolean =
-    // Timeout for NO_RESPONSE is wrapped in a WebClientRequestException
-    it is ReadTimeoutException || it.cause is ReadTimeoutException ||
-      it is ConnectTimeoutException || it.cause is ConnectTimeoutException ||
-      it is SocketException || it.cause is SocketException ||
-      it is PrematureCloseException || it.cause is PrematureCloseException
+  private fun isTimeoutException(e: Throwable): Boolean =
+    // Timeouts etc are wrapped in a WebClientRequestException
+    e is ReadTimeoutException || e.cause is ReadTimeoutException ||
+      e is ConnectTimeoutException || e.cause is ConnectTimeoutException ||
+      e is SocketException || e.cause is SocketException ||
+      e is PrematureCloseException || e.cause is PrematureCloseException || // server closed the connection
+      e is Errors.NativeIoException || e.cause is Errors.NativeIoException // Connection reset by peer
```

Comment:
This exception list is getting messy - there is a case for retrying regardless of exception type possibly